src/compiler/utilities.ts

```diff
@@ -491,7 +491,7 @@ namespace ts {
         }
 
         if (includeJsDoc && hasJSDocNodes(node)) {
-            return getTokenPosOfNode(node.jsDoc![0]);
+            return getTokenPosOfNode(node.jsDoc![0], sourceFile);

```

Comment:
I feel like this function should be changed so that the `sourceFile` param is instead `sourceFile: SourceFileLike | undefined`. That would have probably prevented this bug as someone would likely not do `getTokenPosOfNode(node.jsDoc![0], undefined)`.

Reply:
If you make this change locally and look at broken call sites, do any others omit the parameter when it could be supplied?

Reply:
@andrewbranch the internal `rangeOfNode` function in *compiler/utilities.ts* does not provide a source file, but that's only used by the type checker so, as you know, the nodes will already have a parent then. Probably best for that function to accept a source file as well though.

Reply:
I’ve personally been burned by code that fails to supply a source file to these kinds of functions, both in code I’ve written and in code I’ve consumed, so I’m generally in favor of making this mistake harder to make. But, this is a pattern that parallels a lot of the _public_ API too, which seems unlikely to change, so 🤷‍♀